src/hiccup/form.clj

```diff
@@ -74,6 +74,19 @@
              :value value
              :checked checked?}]))
 
+;; According to the HTML spec, the value inside the boolean `selected` attribute
+;; can only be "" or "selected" (case insensitive).  Hence, the coercion to
+;; boolean is required (otherwise, the attribute value will be `x` itself).
+;; We also need the `(not (sequential? selected))` requirement in the
+;; second `and` because if `selected` is a vector where `x` is not an element,
+;; the first `and` will fail, and the second will try to run something
+;; like `([a b] c)`, and can easily lead to java.lang.IndexOutOfBoundsException.
+(defn- selected? [x selected]
+  (or
+    (and (sequential? selected) (boolean ((set selected) x)))
```

Comment:
Why not place `boolean` outside the `or`? That way it would also guard against `selected` values that are functions.

Reply:
I think your `cond` suggestion is better at conveying intent, so I will use that.  I'll also put `boolean` outside.